scanner/lexer.go

```diff
@@ -519,7 +538,38 @@ func (l *Lexer) newToken(chars []lex.Char) t.Token {
 	startPos := int(firstChar.Pos())
 	endPos := int(lastChar.Pos())
 
-	return t.NewToken(l.charsToBytes(chars), startLine, endLine, startPos, endPos).SetComments(l.Comments)
+	charsBufSize := len(chars) * utf8.UTFMax
+
+	var str string
+
+	if charsBufSize >= initBufSize {
+		runes := make([]rune, 0, len(chars))
+		for _, c := range chars {
+			runes = append(runes, c.Rune)
+		}
+		str = string(runes)
+	} else {
+		var buf *bufStr
+		for {
```

Comment:
It is a bad idea getting objects from Pool until it returns a suitable object.

Reply:
I don’t need these objects because they are nearly full so I can just throw them away. I check that I will eventually find a suitable by checking chars buf size.